### PR TITLE
Data dir bug

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

**Comment by MatusKysel** [comment]
Please do not merge this, wait 'till this issue is tested

**Review by JakubFornadel — APPROVED** [review]

File: libraries/plugins/chain/chain_plugin.cpp [hunk 1]
```diff
@@ -432,16 +420,27 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
...
 
-   my->start_write_processing();
-   if(my->shared_memory_dir.generic_string() == "")
+   if(my->shared_memory_dir.generic_string() == "") {
       my->shared_memory_dir = app().data_dir() / chain_id.str() / "blockchain";
+   }
+
+   //correct directories
+   if( ! my->genesis.is_private_net && bfs::exists( app().data_dir() / "blockchain" ) ){
+      bfs::create_directories ( my->shared_memory_dir );
```
Comment (JakubFornadel) [inline libraries/plugins/chain/chain_plugin.cpp:430]: I would delete "! my->genesis.is_private_net &&" from this condition. But its not a must :)

Comment (ejossev) [reply libraries/plugins/chain/chain_plugin.cpp:430]: I was assuming that sophiatx_app_data/blockchain is used for default, i.e. mainnet net.

**Review by ejossev — APPROVED** [review]

File: libraries/plugins/chain/chain_plugin.cpp (hunk 1 above)
Comment (JakubFornadel) [reply libraries/plugins/chain/chain_plugin.cpp:430]: but according to changes, this condition is always entered(does not matter if it is private net or not):
if(my->shared_memory_dir.generic_string() == "") {
      my->shared_memory_dir = app().data_dir() / chain_id.str() / "blockchain";
   }

so even mainnet is going to use sophiatx_app_data/chain_id/blockchain directory

Comment (ejossev) [reply libraries/plugins/chain/chain_plugin.cpp:430]: Yes, for private nets, the appropriate directory is created, but only for mainnet the old data are migrated from sophiatx_app_data/blockchain to the new directory

Comment (JakubFornadel) [reply libraries/plugins/chain/chain_plugin.cpp:430]: ok, get your point now :)

**Pull request merged**